File: testdata/ts/sample.ts
function greet(name: string): string {
  return "Hello, " + name;
}

const add = (a: number, b: number): number => {
  return a + b;
};

interface Shape {
  area(): number;
}

type Point = {
  x: number;
  y: number;
};

class Circle implements Shape {
  constructor(private radius: number) {}
  area(): number {
    return Math.PI * this.radius ** 2;
  }
}

enum Direction {
  Up,
  Down,
  Left,
  Right,
}

File: testdata/kt/sample.kt
fun greet(name: String): String {
    return "Hello, $name!"
}

fun add(x: Int, y: Int): Int = x + y

data class Point(val x: Double, val y: Double)

class Circle(private val radius: Double) {
    fun area(): Double {
        return Math.PI * radius * radius
    }

    fun perimeter(): Double = 2 * Math.PI * radius
}

interface Shape {
    fun area(): Double
    fun perimeter(): Double
}

object MathUtils {
    fun square(n: Int): Int = n * n
}

enum class Direction {
    UP, DOWN, LEFT, RIGHT
}
